packages/plugins/i18n/server/src/services/ai-localizations.ts

```diff
@@ -193,27 +201,36 @@ const createAILocalizationsService = ({ strapi }: { strapi: Core.Strapi }) => {
           })
         );
       } catch (error) {
+        await aiLocalizationJobsService.upsertJobForDocument({
+          documentId,
+          contentType: model,
+          sourceLocale: document.locale,
+          targetLocales,
+          status: 'failed',
+        });
         strapi.log.error('AI Localizations generation failed', error);
       }
     },
     setupMiddleware() {
       strapi.documents.use(async (context, next) => {
         const result = await next();
 
-        // Only trigger on create/update actions
+        // Only trigger on create/update/publish actions
         if (!['create', 'update'].includes(context.action)) {
           return result;
         }
```

Comment:
the comment adds publish but not the code. I think publish may be mentioned in the PRD but I don't think there's anything to actually translate upon publish